src/asyncUtils.ts

```diff
@@ -80,40 +91,22 @@ function asyncUtils(addResolver) {
     }
   }
 
-  const waitForValueToChange = async (selector, options = {}) => {
+  const waitForValueToChange = async (selector: () => unknown, options = {}) => {
```

Comment:
I think `options` should be typed to `WaitOptions` here to have better support for consumers.